staging/src/k8s.io/dynamic-resource-allocation/controller/controller.go

```diff
@@ -815,7 +823,19 @@ func (ctrl *controller) syncPodSchedulingContexts(ctx context.Context, schedulin
 				})
 			modified = true
 		} else if stringsDiffer(schedulingCtx.Status.ResourceClaims[i].UnsuitableNodes, delayed.UnsuitableNodes) {
-			// Update existing entry.
+			// Update existing entry. We might have checked "potential nodes + selected node" above,
+			// so this list might be too long. When truncating it, make sure that the selected
+			// node is listed.
+			lenUnsuitable := len(delayed.UnsuitableNodes)
+			if lenUnsuitable > resourcev1alpha2.PodSchedulingNodeListMaxSize {
+				if delayed.UnsuitableNodes[0] == selectedNode {
+					// Truncate at the end and keep selected node in the first element.
+					delayed.UnsuitableNodes = delayed.UnsuitableNodes[0:lenUnsuitable-1]
```

Comment:
This truncation assumes that the difference between `lenUnsuitable` and `PodSchedulingNodeListMaxSize` is 1. Is this guaranteed?

Reply:
There was no truncation earlier. Drivers had to return at most `PodSchedulingNodeListMaxSize`, which they did by iterating over the potential nodes slice. Now that slice is potentially one element longer, so truncating by one element works.

The `UnsuitableNodes` method call description should get extended to cover this.